popsycle/synthetic.py

```diff
@@ -3515,32 +3668,400 @@ def generate_slurm_scripts(slurm_config_filename, popsycle_config_filename,
         print('Error: specified number of cores exceeds limit. Exiting...')
         return None
 
+    optional_cmds = ''
+
     # Pass along optional parameters if present
     if overwrite:
-        overwrite_cmd = '--overwrite'
-    else:
-        overwrite_cmd = ''
+        optional_cmds += '--overwrite '
 
-    if seed:
-        seed_cmd = '--seed=%i' % seed
-    else:
-        seed_cmd = ''
+    if seed is not None:
+        optional_cmds += '--seed=%i ' % seed
+
+    if skip_galaxia:
+        optional_cmds += '--skip-galaxia '
+
+    if skip_perform_pop_syn:
+        optional_cmds += '--skip-perform-pop-syn '
+
+    if skip_calc_events:
+        optional_cmds += '--skip-calc-events '
+
+    if skip_refine_events:
+        optional_cmds += '--skip-refine-events '
 
     # Populate the mpi_template specified inputs
     job_script = slurm_template.format(**locals())
 
     # Write the script to the path_run folder
-    script_filename = path_run + '/run_popsycle_%s_%s.sh' % (jobname,
-                                                             output_root)
+    script_filename = path_run + '/run_popsycle_%s.sh' % (jobname)
     with open(script_filename, 'w') as f:
         f.write(job_script)
 
     # Submit the job to disk
     if submitFlag:
+        os.chdir(path_run)
         stdout, stderr = execute('sbatch {0}'.format(script_filename))
-        print('** Standard Out **')
+        print('Submitted job {0} to {1} for {2} time'.format(script_filename,
+                                                             resource,
+                                                             walltime))
+        print('---- Standard Out')
         print(stdout)
-        print('** Standard Err **')
+        print('---- Standard Err')
         print(stderr)
+        print('')
+
+
+def generate_ubv_to_ztf_grid(iso_dir, filter_name):
+    """
+    Creates the 2D transformational matrix `ubv_to_ztf-r_grid.npz' and
+    `ubv_to_ztf-g_grid.npz' necessary for generating ztf-g and ztf-r
+    magnitudes from the UBV filters
+
+    ubv-to-ztf-g
+        x-axis : ubv_v - ubv_r
+        y-axis : ubv_b - ubv_v
+        z-axis : ubv_v - ztf_g
+
+    ubv-to-ztf-r
+        x-axis : ubv_v - ubv_r
+        y-axis : ubv_b - ubv_v
+        z-axis : ubv_r - ztf_r
+
+    Parameters
+    ----------
+    iso_dir : filepath
+        Where are the isochrones stored (for PopStar)
+
+    filter_name : str
+        The name of the filter in which to calculate all the
+        microlensing events. Must be either 'g' or 'r'.
+
+    Output
+    ------
+    None
+
+    """
+
+    # Define isochrone parameters
+    logAge = np.log10(8 * 10 ** 9)  # Age in log(years)
+    # dist = 4000  # distance in parsec
+    dist = 10  # distance in parsec
```

Comment:
I think this is kpc?

Reply:
Oops... I get it. it is parsec (so ignore my comment). Just add a comment that says "calculating absolute magnitudes".

Reply:
Done